### PR TITLE
Changes to make OctoClient modular

### PR DESCRIPTION
**Pull request by sahilverma0696**
Taken changes from https://github.com/finbox-in/octoclient/pull/1
Publishing from variable params to structs

### CODE REVIEW

File: octoclient.go [hunk 1]
```diff
@@ -27,46 +34,51 @@ type OctoResponse struct {
...
-	// TODO: clientID will be replaced with token in coming future
-	apiEndpoint := "/service/invoke"
-	callingUrl := o.BaseURL + apiEndpoint
-	method := "POST"
-	contentType := "application/json"
+	callingUrl := o.baseURL + apiEndpoint
 	var response OctoResponse
 
-	flag := IsValidID(payload.ServiceID.String())&& IsValidID(clientID)
+	flag := IsValidID(payload.ServiceID.String()) && IsValidID(o.clientID.String())
```
Comment (kushalhalder) [inline octoclient.go:66]: Let's remove all validations.

File: octoclient.go [hunk 2]
```diff
@@ -27,46 +34,51 @@ type OctoResponse struct {
...
+		clientID:   clientID,
 	}
 }
 
 func (o *OctoClient) getHttpClient() http.Client {
 	return *o.HTTPClient
 }
 
-func (o *OctoClient) ServiceInvoke(clientID string, payload OctoPayload) (OctoResponse, error) {
+func (o *OctoClient) ServiceInvoke(payload OctoPayload) (OctoResponse, error) {
```
Comment (kushalhalder) [inline octoclient.go:61]: `*OctoResponse` Let's return the pointer, and replace return statements with `nil, err`

File: octoclient.go [hunk 3]
```diff
@@ -27,46 +34,51 @@ type OctoResponse struct {
 
 type OctoClient struct {
 	HTTPClient *http.Client
-	BaseURL    string
+	baseURL    string
+	clientID   uuid.UUID
```
Comment (kushalhalder) [inline octoclient.go:38]: `clientID` to `token`

File: octoclient.go (hunk 2 above)
Comment (kushalhalder) [reply octoclient.go:61]: Also, add context

Comment (sahilverma0696) [reply octoclient.go:61]: make an inline func to give response body
if ctx is null 
Context.Todo()

**Review by rajil36 — APPROVED** [review]

**Comment by sahilverma0696** [comment]
alt text

**Pull request merged**